docs/data/toolpad/core/components/dashboard-layout/DashboardLayoutNavigationActions.js

```diff
@@ -78,60 +93,45 @@ function DashboardLayoutNavigationActions(props) {
   // Remove this const when copying and pasting into your project.
   const demoWindow = window !== undefined ? window() : undefined;
 
+  const popoverMenuAction = (
```

Comment:
I would recommend using tabs on this demo since it becomes very long to read through if expanded

Reply:
> You could perhaps benefit from some code sharing as well.

I'm not sure demos should do much code sharing. Ideally they're self-contained units. And as simple as possible. i.e. as simple as can be to convey the aspect that is being demoed.

Reply:
Agree with ideally them being self-contained, but perhaps we need to optimise for readability if they grow too long?

Reply:
Advantage of single file demoes is that they can be simply copy+pasted from in one action (explains some of the popularity of shadcn). Splitting them across multiple files requires more context switching on behalf of the user to get the example integrated in their own code base. 
I disagree that splitting up in multiple files increases readability, to me it mostly creates extra friction to navigate the code. Instead we should think more in terms of "what can we remove from the code to get to the essence?" and focus on that to increase readability.
I believe the feature of multiple files in demoes should be used very sparingly. i.e. only when the split is mandated by the runtime (_document.js, _app.js) or perhaps to hold a demo dataset.

Reply:
I think if we're optimising for copy and paste-ability, we should ideally: advertise it, and make the demo code blocks much larger. Given the size of our demo code blocks, I think the sense that this is meant to be copy and pasted doesn't immediately come out. (For instance, https://ui.shadcn.com/blocks as a benchmark has "Copy and Paste" as a verb in the header, and a very large code block). 

| Shadcn Blocks | Us |
|--------|--------|
| <img width="1490" alt="Screenshot 2024-08-01 at 5 58 57 PM" src="https://github.com/user-attachments/assets/65d46c6e-c2f3-4fad-92f3-7d5c23c885e9"> | <img width="1503" alt="Screenshot 2024-08-01 at 5 58 26 PM" src="https://github.com/user-attachments/assets/173f70ae-93fa-405c-9edb-13e7af25710a"> | 

One possible solution is to come up with a flag that can render the MUI Docs Layout with navigation hidden, so that code can be much wider.

Within the context of our docs, I think that longer code can become harder for skimming. This is my opinion, okay to drop this if single files being okay for readability is the more popular opinion 👍🏻

Reply:
About this in specific, I tried to make the preview content as short as possible while still showcasing each scenario, but when intentionally expanded I feel like it's ok for things to be longer as long as most of the code is relevant to the example.

Reply:
> One possible solution is to come up with a flag that can render the MUI Docs Layout with navigation hidden, so that code can be much wider.

Yeah, seems like in some of these pages we might benefit from taking the full-width of the page as shadcn does, but so far I guess it hasn't been super necessary.

Reply:
> I think if we're optimising for copy and paste-ability, we should ideally: advertise it, and make the demo code blocks much larger.

We don't optimize for copy and paste-ability in an absolute sense. Definitely not the demo previews. They are too small for most demos to achieve that, they are to help the user decide whether the particular demo from a code point of view is going to be applicable to their problem. You could say the previews are mostly optimized for scanneability, that's why they should be short, to the point and not scrollable. If such a preview is not possible, then there should be likely just no preview. The expanded version should be more optimized for copy-pasteability. 

shadcn is different a different case, they basically act as a snippets library. Our demos are not snippets, we show the integration of our libraries. When I copy+paste from our docs I usually only copy portions, not the whole demo.